docs/API_v1.0.md

```diff
@@ -1,174 +1,408 @@
-| --------- | -----------------------------------  | ---------------------------- | ------------------ |
-| GET       | /api/v1.0/poll/{pollId}/subscription | Get subscription status      | 200, 403, 404      |
-| PUT       | /api/v1.0/poll/{pollId}/subscription | Subcribe                     | 201, 403           |
-| DELETE    | /api/v1.0/poll/{pollId}/subscription | unsubscribe                  | 200, 403           |
+
+# DRAFT
+
+This documentation and the API are not final and may contain issues and bugs!
+Details may change!
+
+To act like a authenticated user, use username:password@ prefix in the URL
+Parameters in the body override the URL-parameter
+
+Base URL for all calls: /index.php/apps/polls
+Example calls:
+* Gets all comments of poll no. 1
+`https://username:password@nextcloud.local/index.php/apps/polls/api/v1.0/poll/1/comments`
+```bash
+`curl -u username:password -X GET https://nextcloud.local/index.php/apps/polls/api/v1.0/poll/1/comments`
+```
+* Create a new poll
+```bash
+`curl  -u username -X POST https://nextcloud.local/index.php/apps/polls/api/v1.0/poll -H "Content-Type: application/json;charset=utf-8" -d "{\"title\": \"New poll\", \"type\": \"datePoll\"}"`
+```
+
+# Poll
+## Default functions
+| Method    | Endpoint                 | Payload | Description            | Return codes       | Return value   |
+| --------- | ------------------------ | ------- | ---------------------- | ------------------ | -------------- |
+| GET       | /api/v1.0/polls          | no      | Get array of polls     | 200, 403, 404      | array          |
+| GET       | /api/v1.0/poll/{pollId}  | no      | Get poll with {pollId} | 200, 403, 404      | requested poll |
+| POST      | /api/v1.0/poll           | yes     | Add new poll           | 201, 403, 404      | added poll     |
+| PUT       | /api/v1.0/poll/{pollId}  | yes     | Update poll            | 200, 403, 404, 409 | updated poll   |
+| DELETE    | /api/v1.0/poll/{pollId}  | no      | Delete poll            | 200, 403, 404      | deleted poll   |
+
+
+## Special functions
+| Method    | Endpoint                      | Payload | Description                  | Return codes       | Return value   |
+| --------- | ------------------------------| ------- | ---------------------------- | ------------------ | -------------- |
+| POST      | /api/v1.0/poll/{pollId}/clone | no      | Clone poll from {pollId}     | 201, 403, 404      | cloned poll    |
+| POST      | /api/v1.0/poll/{pollId}/trash | no      | Move to/remove from trash    | 200, 403, 404      | updated poll   |
+| GET       | /api/v1.0/enum/poll           | no      | Get valid enums              | 200, 403, 404      | array          |
+
+## Valid payloads
+### Add new poll
+```json
+{
+    "type": "datePoll",
+    "title": "Poll Title"
+}
+```
+
+### Update poll
+Send the full or a partial structure.
+In this example, we set an expiration date on January 1, 1970 at 00:00:01 UTC.
```

Comment:
Additionally it could be mentioned, that an expiry timestamp in the past represents a closed poll, 0 represents an endless poll (aka without expiration) and an expiry timestamp in the future sets an automatic closing date. So this example closes the poll.